lib/screens/input_password.dart

```diff
@@ -1,13 +1,93 @@
 import 'package:flutter/material.dart';
+import 'package:niira/models/game.dart';
+import 'package:niira/models/player.dart';
+import 'package:niira/screens/waiting_for_game_to_start.dart';
+import 'package:niira/services/auth/auth_service.dart';
+import 'package:niira/services/database/database_service.dart';
+import 'package:provider/provider.dart';
+
+class InputPasswordScreen extends StatefulWidget {
+  final Game game;
+  InputPasswordScreen({@required this.game});
+  @override
+  _InputPasswordScreenState createState() => _InputPasswordScreenState();
+}
+
+class _InputPasswordScreenState extends State<InputPasswordScreen> {
+  final _formKey = GlobalKey<FormState>();
 
-class InputPasswordScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       key: Key('inputPasswordScreen'),
+      appBar: AppBar(
+        automaticallyImplyLeading: true,
+        title: Text('Password'),
+      ),
+      floatingActionButton: FloatingActionButton.extended(
+        key: Key('input_password_screen_submit_btn'),
+        label: Text(
+          'Next',
+          style: TextStyle(color: Colors.white),
+        ),
+        icon: Icon(
+          Icons.navigate_next,
+          color: Colors.white,
+        ),
+        onPressed: () async {
+          if (_formKey.currentState.validate()) {
+            // get player details
+            final userId = await context.read<AuthService>().currentUserId;
+            final username =
+                await context.read<DatabaseService>().getUserName(userId);
+
+            // add user as player to game
+            final player = Player(
+              id: userId,
+              username: username,
+              isTagger: false,
+              hasBeenTagged: false,
+              hasItem: false,
+              isAdmin: false,
+            );
+            await context
+                .read<DatabaseService>()
+                .joinGame(widget.game.id, player);
+
+            // navigate to waiting screen
+            await Navigator.push<dynamic>(
```

Comment:
Since we last spoke about this I've been thinking about it a fair bit as I was working on the same thing in another context - my opinion has changed and I think it's really worth using the NavigationService whenever we navigate. I wouldn't suggest it blocks this PR but wanted to flag it so we can talk about it at some stage.

Reply:
sweet ill make a note in the project!